src/main/java/org/logstash/beats/BatchEncoder.java

```diff
@@ -0,0 +1,81 @@
+package org.logstash.beats;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToByteEncoder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Map;
+
+
+/**
+ *  This Class is mostly used in the test suite to make the right assertions with the encoded data frame.
+ *  This class support creating v1 or v2 lumberjack frames.
+ *
+ */
+public class BatchEncoder extends MessageToByteEncoder<Batch> {
+    private final static Logger logger = LogManager.getLogger(BatchEncoder.class.getName());
+
+
+    @Override
+    protected void encode(ChannelHandlerContext ctx, Batch batch, ByteBuf out) throws Exception {
+        out.writeByte(batch.getProtocol());
+        out.writeByte('W');
+        out.writeInt(batch.size());
+
+        ByteBuf buffer = getPayload(ctx, batch);
+
+        try {
+            out.writeBytes(buffer);
+        } finally {
+            buffer.release();
+        }
+    }
+
+    protected ByteBuf getPayload(ChannelHandlerContext ctx, Batch batch) throws IOException {
+        ByteBuf payload = ctx.alloc().buffer();
```

Comment:
indeed, not sure if a useful approximation exists, messages can be very different in size. ignore this comment